vmm/src/vm.rs

```diff
@@ -609,27 +609,22 @@ impl Vm {
                 .map_err(Error::InitializeTdxVm)?;
         }
 
-        vm.init().map_err(Error::InitializeVm)?;
-
-        cpu_manager
-            .lock()
-            .unwrap()
-            .create_boot_vcpus(snapshot_from_id(snapshot.as_ref(), CPU_MANAGER_SNAPSHOT_ID))
-            .map_err(Error::CpuManager)?;
-
-        // This initial SEV-SNP configuration must be done immediately after
-        // vCPUs are created. As part of this initialization we are
-        // transitioning the guest into secure state.
-        #[cfg(feature = "sev_snp")]
-        if sev_snp_enabled {
-            vm.sev_snp_init().map_err(Error::InitializeSevSnpVm)?;
-        }
-
         #[cfg(feature = "tdx")]
         let dynamic = !tdx_enabled;
         #[cfg(not(feature = "tdx"))]
         let dynamic = true;
 
+        #[cfg(feature = "kvm")]
+        let is_kvm = matches!(
+            hypervisor.hypervisor_type(),
+            hypervisor::HypervisorType::Kvm
+        );
+        #[cfg(feature = "mshv")]
+        let is_mshv = matches!(
+            hypervisor.hypervisor_type(),
+            hypervisor::HypervisorType::Mshv
+        );
+
```

Comment:
A bigger change (for the future) - maybe we could establish a new pattern with something like `Hypervisor::hypervisor_type_matches(HypervisorType) -> bool` and make the `HypervisorType` members non conditional. Might clean up a lot of the spaghetti that comes from the conditional code.

Reply:
Yeah I can take that up in a follow up PR.